http_request_randomizer/requests/proxy/requestProxy.py

```diff
@@ -99,7 +99,7 @@ def generate_proxied_request(self, url, method="GET", params={}, data={}, header
 
             self.logger.debug("Using proxy: {0}".format(str(self.current_proxy)))
             request = requests.request(method, url, proxies={"http": self.current_proxy},
-                                       headers=headers, data=data, params=params, timeout=req_timeout)
+                                       headers=headers, data=data, params=params, timeout=req_timeout, allow_redirects=False)
```

Comment:
In the _tests/mocks.py_  file  **samair_mock** content should be identical to what you see in the real web page. So when you run the test_providers all the tests should pass.

No worries at all, push your changes once you are ready and I will pick it up from there.

Cheers,
Panagiotis